### ISSUE TITLE
[bug] PST : Spell Visuals: Unimplemented Front/behind animations

### ISSUE THREAD

**Issue by chilvence** [gemrb/gemrb#639]
Several casting and effect animations have two cycles that are meant to be played simultaneously, with one in front of the sprite and one behind. Currently Gemrb plays them one after the other instead. 

![grb scripture casting glow](https://user-images.githubusercontent.com/1923867/76098931-3e324f00-5fc2-11ea-8483-556e1450a835.png)

![pst scripture casting glow](https://user-images.githubusercontent.com/1923867/76098998-5a35f080-5fc2-11ea-8868-d379415eb00f.png)

This is the second issue pulled from #587 that affects a large swathe of the spells in the game. It is only really visual, but it does really break the illusion that anything is working properly because it affects most of the casting glows.

Here is a complete list of animations that need this treatment drawn by sifting through the game data:

 - Casting glows that are referenced by number in the spell file

   - A001ABJN		: Abjuration cast glow
   - A002AALT 		: Alteration cast glow
   - A004ENCH 	: Enchantment cast glow
   - A006ILLU 		: Illusion cast glow
   - A008CNJR 		: Conjuration cast glow
   - A009NECR 	: Necromancy cast glow
   - A009SKIL 		: Skill use cast glow 

Spells that have similar acting bam files and the locations where they referenced (counting from 0 for OCD reasons)

- SPWI108 Pacify > Ability 0  > # 1 Op 187 > 
  - S013STAR 		: Pacify caster effect

- SPWI212 Swarm Curse > Global Effects 0, 1 > Op 188 >
  - S030SWEN 		: Swarm curse encircle
  - S030SWGR 		: Swarm curse grow

- SPWI304 Cloak of Warding > Ability 0  > # 0 Op 187 >
  - S034CLWD : Cloak visual effect

- SPWI405 Force Missile > All Abilities > # 2 Op 191 >
  - S045DFRC : Force missile hit

- SPWI305 & SPWI313 Elysiums tears / Fiery rain > Ability 0 > # 2 Op 191 >
  - S070CSTR : Caster magic rising effect

- SPWI704 Guardian Mantle > Ability 0 > # 0 Op 187 >
  - S074SHRD: Mantle appearance 

- SPWI905 Elysiums Fires > Ability 0 > # 1 Op 191 >
  - S095CAST : Fire caster effect

 - SPPR102 Cure Light wounds > Ability 0 > # 1 Op 187 >
   - S0P2CUR3 : Positive energy effect

 - Location reference unkown ???? : 
   - S025COLD : Ice knife projectile hit

 - The following animations don't just have 2 cycles for front or back, they have three sets of 2 in the same BAM file (counting from 1 instead because NI does...)
   - 1-2: for effect appearance
   - 3-4: for looping while active 
   - 5-6: for ending of effect

 - SPWI504 Enoll’s duplication > Ability 0 > # 1  Op 201 > 
   - S052RING : Spell visually active effect : circling rings

 - SPWI203 Black Barbed Shield > Ability 0 > # 0 Op 201 >
   - S023BBSH : Spell active effect: spinning barbs

 - SPWI603: Howl of Pandemonium > Ability 0 > # 0 Op 187 > 
   - S055SKLC : Spell victim effect: circling skull	

 - Location Reference Unkown ???? :
   - S055SWHD : Unused / Harcoded related to SPWI603 ?

**Comment by lynxlynxlynx** [comment]
Which spell is in the screenshots?

**Comment by chilvence** [comment]
That is the generic conjuration glow (# 37), but I think the particular spell I used was scripture of steel (SPWI118)

**Comment by lynxlynxlynx** [comment]
Did it perchance work before 48e20b5? Just to see if that regression mentioned in #628 is related or not.

**Comment by chilvence** [comment]
I tried with 22b318e, It appears to be an unrelated problem. I don't remember these animations working at any point, I had just ignored them before as a nuisance rather than a serious problem.

I notice that BG2 has separate spaurafb / spauraff  bam files for doing a similar encirclement effect, also spblbtop / spblbbtm. This appears handled by two separate VVC files, with the lower one offset on the y axis. They are basically called separately by the SPL file and apparently the coordinates take care of everything being drawn in the correct back to front order.

 But then it has spfiresb / spfiresr bam files that have the back front cycles in one file, and the two separate vvc files for each one just reference the different animation cycles.

PST doesn't do any of this, it just has one bam file for each front back animation called by a single op, and no VVC support, although could it be useful that gemrb allows it to solve this?

Tellingly, if I try to swap any of these animations, they don't play correctly any more, eg if I copy A002AALT.bam to A005LDIV.bam, whatever code is deciding which casting glow to use only bothers to load the front side of the animation. They are also all clearly referenced by name in the executable, though I am still not able to read machine code very well :)

**Comment by chilvence** [comment]
Another unexpected thing is that at least one of these effects is partially handled

![s013star](https://user-images.githubusercontent.com/1923867/76684115-2cbff700-6601-11ea-8074-1b1809a3e45a.png)

Which is the flash effect from the pacify spell


I also forgot to add, I've since learned that several of these spells are not actually accessible without cheating anyway: howl of pandemonium, elysium's fires, guardian mantle were all cut from the game for some reason. So that's three less spells to worry about.

**Comment by lynxlynxlynx** [comment]
Yes, we can try with vvcs; perhaps the loader needs adapting, but it'd be the same principle as elsewhere that we successively try different file formats for the same resref.

**Comment by lynxlynxlynx** [comment]
Looking at casting glows — pst uses twin graphics, there's one cycle per each bottom/top in the bam. **We already handle that and also display both**, but currently for this use case, **they're not played simultaneously**. 

spwi805 powerword blind is my test case. It's enough to starting look at code from fx_casting_glow onwards. Toggling the parameter to LoadAnimationFactory looks like a correct thing to do, but it only makes the bottom animation play twice instead. So it'd be good to find some twin bam elsewhere that's already working properly, to be able to check which assumptions break for casting glows.
